pubsub/hazelcast/hazelcast.go

```diff
@@ -0,0 +1,108 @@
+package hazelcast
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/dapr/dapr/pkg/logger"
+	hazelcastCore "github.com/hazelcast/hazelcast-go-client/core"
+	"strings"
+
+	"github.com/dapr/components-contrib/pubsub"
+
+	"github.com/hazelcast/hazelcast-go-client"
+)
+
+const (
+	hazelcastServers = "hazelcastServers"
+)
+
+type Hazelcast struct {
+	client hazelcast.Client
+	logger logger.Logger
+}
+
+// NewHazelcastPubSub returns a new hazelcast pub-sub implementation
+func NewHazelcastPubSub(logger logger.Logger) pubsub.PubSub {
+	return &Hazelcast{logger: logger}
+}
+
+func parseHazelcastMetadata(meta pubsub.Metadata) (metadata, error) {
+	m := metadata{}
+	if val, ok := meta.Properties[hazelcastServers]; ok && val != "" {
+		m.hazelcastServers = val
+	} else {
+		return m, errors.New("hazelcast error: missing hazelcast servers")
+	}
+	return m, nil
+}
+
+func (p *Hazelcast) Init(metadata pubsub.Metadata) error {
+	m, err := parseHazelcastMetadata(metadata)
+	if err != nil {
+		return err
+	}
+
+	hzConfig := hazelcast.NewConfig()
+
+	servers := m.hazelcastServers
+	hzConfig.NetworkConfig().AddAddress(strings.Split(servers, ",")...)
+
+	p.client, err = hazelcast.NewClientWithConfig(hzConfig)
+	if err != nil {
+		return fmt.Errorf("hazelcast error: failed to create new client, %v", err)
+	}
+
+	return nil
+}
+
+func (p *Hazelcast) Publish(req *pubsub.PublishRequest) error {
+	topic, err := p.client.GetTopic(req.Topic)
+	if err != nil {
+		return fmt.Errorf("hazelcast error: failed to get topic for %s", req.Topic)
+	}
+
+	if err = topic.Publish(req.Data); err != nil {
+		return fmt.Errorf("hazelcast error: failed to publish data, %v", err)
+	}
+
+	return nil
+}
+
+func (p *Hazelcast) Subscribe(req pubsub.SubscribeRequest, handler func(msg *pubsub.NewMessage) error) error {
+	topic, err := p.client.GetTopic(req.Topic)
+	if err != nil {
+		return fmt.Errorf("hazelcast error: failed to get topic for %s", req.Topic)
+	}
+
+	_, err = topic.AddMessageListener(&hazelcastMessageListener{topic.Name(), handler})
+	if err != nil {
+		return fmt.Errorf("hazelcast error: failed to add new listener, %v", err)
+	}
+
+	return nil
+}
+
+type hazelcastMessageListener struct {
+	topicName     string
+	pubsubHandler func(msg *pubsub.NewMessage) error
+}
+
+func (l *hazelcastMessageListener) OnMessage(message hazelcastCore.Message) error {
+	var msg []byte
+	m, ok := message.MessageObject().([]byte)
+	if ok {
+		msg = m
+	} else {
+		msg, _ = json.Marshal(message.MessageObject())
```

Comment:
Why do you need this fallback to parse the message? I would expect for all messages to have the save structure.

Reply:
I looked a second time. You right, I actually dont need it

Reply:
@peterjot once you make that change, we can merge this PR. otherwise it looks good to me.